rest/config.go

```diff
@@ -284,6 +288,68 @@ type Http2Config struct {
 	Enabled *bool `json:"enabled,omitempty"` // Whether HTTP2 support is enabled
 }
 
+//TODO: Add support for TLS 1.3 when we run Go 1.13
+type TLSVersion uint32
+
+const (
+	TLSv1 TLSVersion = iota
+	TLSv1dot1
+	TLSv1dot2
```

Comment:
These constants are already defined in the TLS library - in general I feel like defining a new TLSVersion type may be over-architecting the problem.  I'm not sure we need anything more than a 'stringToTLSVersion' function to map a string config value to the constants.  Is the motivation to provide additional validation at unmarshal time?

Reply:
Very true... In all honesty I was basing it off of the logging one however I could probably swap this out...

Reply:
Thinking about it more, I definitely think that ServerConfig.TLSVersion should just be a *string.  I don't think performing validation at unmarshal time is useful, given that it limits the ability to report multiple errors in a single config.  I think switching to follow the pattern used for the other *string config values would be preferred (as well as avoiding the const declarations here).